cachi2/core/scm.py

```diff
@@ -42,7 +43,15 @@ def get_repo_id(repo: Union[str, PathLike[str], Repo]) -> RepoID:
     See `man git-clone` (GIT URLS) for some of the url formats that git supports.
     """
     if isinstance(repo, (str, PathLike)):
-        repo = Repo(repo, search_parent_directories=True)
+        try:
+            repo = Repo(repo, search_parent_directories=True)
+        except (InvalidGitRepositoryError, NoSuchPathError):
+            raise UnsupportedFeature(
+                "Cachi2 cannot process provided path as a git repository",
+                solution=(
+                    "Please ensure that the path is correct and that it is a valid git repository."
+                ),
+            )
```

Comment:
This commit is irrelevant to the Makefile changes topic, please propose this in a separate PR for cleaner bookkeeping.